linode/manager
# upcoming: [M3-7520] - AGLB Fixes and Improvements

## Description 📝

- Fixes ActionCell styles 🎨
- Make copy changes required by technical writer ✏️
  - Add rule drawer now has dynamic copy depending on the protocol
- Improves spacing of some error notices ⛔️
- Fixes a lot of validation / payloads to match what the Linode API allows 🚛
  - You should now be able to add TCP Rules now
  - You should be able to delete Rules

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-12-01 at 4 48 00 PM](https://github.com/linode/manager/assets/115251059/07dc9ca4-e4ce-46c3-8170-7340bf87c39a) | ![Screenshot 2023-12-01 at 4 47 52 PM](https://github.com/linode/manager/assets/115251059/93df0a3f-275b-4780-8edc-342fe390a524) |
| ![Screenshot 2023-12-01 at 4 50 18 PM](https://github.com/linode/manager/assets/115251059/15225359-3dd1-4d50-b85b-09287914f1cd) | ![Screenshot 2023-12-01 at 4 50 35 PM](https://github.com/linode/manager/assets/115251059/99fa56f3-efbf-4688-8773-9d82962ce0b5) |
| ![Screenshot 2023-12-04 at 8 48 14 AM](https://github.com/linode/manager/assets/115251059/72f426b3-8b18-401d-8fc3-30f088958e56) | ![Screenshot 2023-12-04 at 8 47 37 AM](https://github.com/linode/manager/assets/115251059/7b512724-aa1d-46cd-a507-f4f8c5e4a394)
|  ![Screenshot 2023-12-04 at 8 48 24 AM](https://github.com/linode/manager/assets/115251059/c39be6ef-5adf-40f8-adef-b22a1a916ab3) | ![Screenshot 2023-12-04 at 8 47 22 AM](https://github.com/linode/manager/assets/115251059/bb132546-3f7c-47c1-a4c7-910b4bc2350d) |





## How to test 🧪

### Prerequisites
- Use MSW to test **or** use the `dev-test-aglb` credentials

### Verification steps 
- Verify the changes listed in the `Description` section 
- Check for any regressions to existing behavior

## As an Author I have considered 🤔
